core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1704,9 +1734,26 @@ object Log {
     * Return a directory name to rename the log directory to for async deletion. The name will be in the following
     * format: topic-partition.uniqueId-delete where topic, partition and uniqueId are variables.
     */
-  def logDeleteDirName(logName: String): String = {
+  def logDeleteDirName(topicPartition: TopicPartition): String = {
+    val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-", "")
+    s"${topicPartition.topic}-${topicPartition.partition}.$uniqueId$DeleteDirSuffix"
+  }
+
+  /**
+    * Return a future directory name for the given topic partition. The name will be in the following
+    * format: topic-partition.uniqueId-future where topic, partition and uniqueId are variables.
+    */
+  def logFutureDirName(topicPartition: TopicPartition): String = {
     val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-", "")
-    s"$logName.$uniqueId$DeleteDirSuffix"
+    s"${topicPartition.topic}-${topicPartition.partition}.$uniqueId$FutureDirSuffix"
+  }
+
+  /**
+    * Return a directory name for the given topic partition. The name will be in the following
+    * format: topic-partition where topic, partition are variables.
+    */
+  def logDirName(topicPartition: TopicPartition): String = {
```

Comment:
We can use this in `logFutureDirName` and `logDeleteDirName`. We can maybe add a `suffix` parameter and have `""` as the default.

Reply:
Sure. I have updated the patch to use this method in the newly-added private method.